pymongo/errors.py

```diff
@@ -167,6 +174,8 @@ def details(self):
         """
         return self.__details
 
+    def __str__(self):
+        return "%s, full error: %s" % (self.__error, self.__details)
```

Comment:
I think we'll need some more work here to specially handle unicode errors on Python 2. Check out the `PyMongoError.__str__/__unicode__` methods for some inspiration. It might be easier to explain in person but [PYTHON-1966](https://jira.mongodb.org/browse/PYTHON-1966) has some more context. FYI addressing this issue will resolve the remaining `test_write_error_unicode` failure.